Update Identity_IdentityProvider.md

digitaljoshua opened the pull request:
Added:

•	GET/api/{version}/tenants/{tenantId}/IdentityProviders/{IdpId}/Users?query={user}&skip={skip}&count={count}&timeout={optional}&skipToken={optional}
•	GET/api/{version}/tenants/{tenantId}/IdentityProviders/{IdpId}/Groups?query={group}&skip={skip}&count={count}&timeout={optional} &skipToken={optional}
•	GET/api/{version}/tenants/{tenantId}/IdentityProviders/{IdpId}/Users/{userId}/Groups?skip={skip}&count={count}&timeout={optional} &skipToken={optional}
•	GET/api/{version}/tenants/{tenantId}/IdentityProviders/{IdpId}/Groups/{GroupId}/Members?skip={skip}&count={count} &skipToken={optional}

Please check the parameters descriptions, values, ranges, and any missing default values. I had to guess.

1. I am unsure on the skip parameter. Are we returning multiple users? Why do we need to skip? Please improve my description
2. what is a skipToken? can you describe better? defaults?
3. Can you describe Timeout better? in seconds? what is defaults?
3. Did i label the json 200 return properly on the four new functions? please check

sukhdev1103 commented on Content_Portal/Documentation/Identity/Identity_IdentityProvider.md:
```diff
@@ -882,6 +882,424 @@ Tenant not found.
 Internal server error.
 ***
 
+## `Get a List of all User on an Identity Provider`
```
I would not say all users because this endpoint would return back only those users that match the query pattern.
query={su} => this would return back the users whose name or email starts with "su".

digitaljoshua replied:
modified to read: Get a list of users that matches the query string on an Identity Provider...

sukhdev1103 commented on Content_Portal/Documentation/Identity/Identity_IdentityProvider.md:
```diff
@@ -882,6 +882,424 @@ Tenant not found.
+
+### Request
+
+`GET api/v1/Tenants/{tenantId}/IdentityProviders/{identityProviderId}/{IdpId}/Users?query={user}&skip={skip}&count={count}&timeout={optional}&skipToken={optional}`
```
we should remove skip and timeout references because they are used on on-prem only.

digitaljoshua replied:
I removed skip and timeout from the syntax and parameters descriptions

sukhdev1103 commented on Content_Portal/Documentation/Identity/Identity_IdentityProvider.md:
````diff
@@ -882,6 +882,424 @@ Tenant not found.
+Id of Identity Provider.
+
+```csharp
+[FromQuery]
````
remove skip

digitaljoshua replied:
removed from syntax

sukhdev1103 commented on Content_Portal/Documentation/Identity/Identity_IdentityProvider.md:
````diff
@@ -882,6 +882,424 @@ Tenant not found.
+
+Maximum number of users to return.
+
+```csharp
````
remove timeout

digitaljoshua replied:
removed timeout description

sukhdev1103 commented on Content_Portal/Documentation/Identity/Identity_IdentityProvider.md:
```diff
@@ -882,6 +882,424 @@ Tenant not found.
+
+### Security
+
+Allowed for these roles:
```
It is only allowed for Account Administrator.

digitaljoshua replied:
Removed the Account Member

sukhdev1103 commented:
1. I am unsure on the skip parameter. Are we returning multiple users? Why do we need to skip? Please improve my description - Pls remove the skip parameter references from the doc. This applies to all the endpoints/routes.
    2. what is a skipToken? can you describe better? defaults? - It's default value is null. 
    Definition: Retrieves the next page of results from result sets that span multiple pages.
For e.g. If there are 100 users [name/email starts with "a"] in AzureAD tenant  and you requested for /users?query=a&count=50 then you will get back 50 users and skiptoken pointing to 51th user. skipToken will be encoded string which is nothing but an index to remaining set of users.
    3. Can you describe Timeout better? in seconds? what is defaults? - Pls remove the skip parameter references from the doc. This applies to all the endpoints/routes.
    4. Did i label the json 200 return properly on the four new functions? please check - seems right!

digitaljoshua commented:
**I did the following for all 4 of the following calls**:
_• GET/api/{version}/tenants/{tenantId}/IdentityProviders/{IdpId}/Users?query={user}&skip={skip}&count={count}&timeout={optional}&skipToken={optional}
• GET/api/{version}/tenants/{tenantId}/IdentityProviders/{IdpId}/Groups?query={group}&skip={skip}&count={count}&timeout={optional} &skipToken={optional}
• GET/api/{version}/tenants/{tenantId}/IdentityProviders/{IdpId}/Users/{userId}/Groups?skip={skip}&count={count}&timeout={optional} &skipToken={optional}
• GET/api/{version}/tenants/{tenantId}/IdentityProviders/{IdpId}/Groups/{GroupId}/Members?skip={skip}&count={count} &skipToken={optional}_

1. removed skip and timeout from syntax of each call
2. removed skip and timeout descriptions from parameter descriptions of each call
3. removed 'Account Member' from the security section of each call

Please double check and verify my changes, Sukhdev. Thanks.

sukhdev1103 commented on Content_Portal/Documentation/Identity/Identity_IdentityProvider.md:
```diff
@@ -882,6 +882,352 @@ Tenant not found.
 
+## `Get a List of all User on an Identity Provider`
+
+Get a list of users that matches the query string on an Identity Provider that supports advanced integration, such as Azure Active Directory. The prerequisite is that the identity provider must have already consented to sharing access to its directory with the OCS tenant.
```
that supports deeper integration ....

sukhdev1103 commented on Content_Portal/Documentation/Identity/Identity_IdentityProvider.md:
````diff
@@ -882,6 +882,352 @@ Tenant not found.
+string skiptoken
+```
+
+Start of returned users after skip count.
````
what is skip count here?

digitaljoshua replied:
I changed the description to: An encoded string that identifies the set of users that was not returned. For example, if you request a count of the first 50 users matching your query, the skiptoken identifies the 51st user.

I made similar modifications to the other 3 API calls as appropriate to them.

sukhdev1103 commented on Content_Portal/Documentation/Identity/Identity_IdentityProvider.md:
```diff
@@ -882,6 +882,352 @@ Tenant not found.
+
+## `Get a List of all Groups on an Identity Provider`
+
+Get a list of all existing groups on an Identity Provider that supports advanced integration, such as Azure Active Directory. The prerequisite is that the identity provider must have already consented to sharing access to its directory with the OCS tenant.
```
we should be more specific about what level of access a customer would consent with the OCS tenant, which are User.Read.All and 
GroupMember.Read.All

sukhdev1103 replied:
> 
> 
> we should be more specific about what level of access a customer would consent with the OCS tenant, which are User.Read.All and
> GroupMember.Read.All

this also applies for users endpoint. Thanks

digitaljoshua replied:
Added the following sentence: The consent grants User.Read.All and GroupMember.Read.all permissions to the OCS tenant.

sukhdev1103 commented on Content_Portal/Documentation/Identity/Identity_IdentityProvider.md:
```diff
@@ -882,6 +882,352 @@ Tenant not found.
+
+## `Get a List of all Groups on an Identity Provider`
+
+Get a list of all existing groups on an Identity Provider that supports advanced integration, such as Azure Active Directory. The prerequisite is that the identity provider must have already consented to sharing access to its directory with the OCS tenant.
```
Get a list of groups that matches the query string on an Identity Provider that supports advanced integration,

digitaljoshua replied:
modified the sentence as suggested.

digitaljoshua commented:
I have made all of the edits related to suggestions and comments from Sukhdev.
